block/fork.go

```diff
@@ -231,16 +252,16 @@ func (m *Manager) createNewBlocks(consensusMsgs []proto.Message) error {
 	return nil
 }
 
-// ensureBatchExists verifies and, if necessary, creates a batch at the specified height.
+// handleForkBatchSubmission verifies and, if necessary, creates a batch at the specified height.
 // This function is critical for maintaining batch consistency in the blockchain while
 // preventing duplicate batch submissions.
 //
 // The function performs the following operations:
 //  1. Checks for an existing batch at the specified height via SLClient
 //  2. If no batch exists, creates and submits a new one
-func (m *Manager) ensureBatchExists(height uint64) error {
+func (m *Manager) handleForkBatchSubmission(height uint64) error {
 	resp, err := m.SLClient.GetBatchAtHeight(height)
-	if err != nil {
+	if err != nil && !errors.Is(err, gerrc.ErrNotFound) {
```

Comment:
how u can handle fork instruction, when there's already state info for this height?

Reply:
in case state info already exists, it means it was already submitted before and there is no need to submit again. i think this will only happen in case the nodes was stopped while submitting this batch

Reply:
how u got into `handleSequencerForkTransition` in this case?
IMO it should have been checked in `m.forkNeeded()`

Reply:
it will enter because the instruction file is not deleted till the batch is submitted (so forkneeded() will be true). so when the node starts again it will enter the handleSequencerForkTransition again, see the block already exists but not the batch and will submit the blocks

Reply:
exactly. documented